File: strings/javascript/prefixo-comum.js
function longestCommonPrefix(v) {
  v.sort();
  let first = v[0];
  let last = v[v.length - 1];
  let result = [];

  for (let i = 0; i < Math.min(first.length, last.length); i++) {
    if (first[i] !== last[i]) {
      return result.join("");
    }
    result.push(first[i]);
  }

  return result.join("");
}

console.log(`"${longestCommonPrefix(["flower", "flow", "flight"])}"`);
console.log(`"${longestCommonPrefix(["dog", "racecar", "car"])}"`);

File: strings/java/src/PrefixoComum.java
import java.util.Arrays;

public class PrefixoComum {

	public static void main(String[] args) {
		System.out.println("\"" + longestCommonPrefix(new String[] { "flower", "flow", "flight" }) + "\"");
		System.out.println("\"" + longestCommonPrefix(new String[] { "dog","racecar","car" }) + "\"");
	}

	public static String longestCommonPrefix(String[] v) {

		StringBuilder result = new StringBuilder();

		Arrays.sort(v);
		String first = v[0];
		String last = v[v.length - 1];

		for (int i = 0; i < Math.min(first.length(), last.length()); i++) {
			if (first.charAt(i) != last.charAt(i)) {
				return result.toString();
			}
			result.append(first.charAt(i));
		}

		return result.toString();
	}
}
